### PR TITLE
Asset group creation

### PR DESCRIPTION
**Pull request by hwindsor**
## Pull Request Overview

- Creation of asset group via post with sightings and encounters data. First step of IA pipeline. 
- Parsing of data can be from incoming request or when read from local git repository. 

---

**REST API Updates [OPTIONAL]**

AssetGroup post now takes the following format 
```
[POST] /api/v1/asset_group
{
    'bulkUpload' : False,
    'description' : 'optional user assigned value'
    'speciesDetectionModel': [{'modelToUse'}],
    'transactionId': '11111111-1111-1111-1111-111111111111',
    'submitterEmail': 'optional email address for anonymous only',
    'sightings': [{
		'startTime': '2021-04-27T08:16:00.901012',
                'context': 'test',
                'name': 'optional free form string assigned by user',
  		'locationId': 'text string',
                'encounters': [{
			'ownerEmail': 'optional email address to assign to'
                        'assetReferences': [{'zebra.jpg'}]
		}]
	}],
}

```

**Pull Request Checklist**
- [x] Ensure that the PR is properly formatted
  - Example: All lint checks are passing
- [x] Ensure that the PR is properly rebased
  - Example: The PR is rebased on `develop` (commit: `<insert develop commit hash>`)
- [x] Ensure that the PR uses a consolidated database migration
  - Example: One database migration is proposed (revision `<insert develop revision>` -> `<insert new revision>`)
- [x] Ensure that the PR is properly tested
  - Example: All automated tests are passing
- [x] Ensure that the PR is properly covered
  - Example: The percentage of the code covered by tests has not decreased
- [x] Ensure that the PR is properly sanitized
  - Example: No sensitive data or large content was added to this PR
- [ ] Ensure that the PR is properly reviewed

### CODE REVIEW

File: tests/extensions/tus/utils.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
```
Comment (hwindsor) [inline tests/extensions/tus/utils.py:1]: No content change, just moved from tests dir

File: tests/modules/asset_groups/resources/test_create_asset_group.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
```
Comment (hwindsor) [inline tests/modules/asset_groups/resources/test_create_asset_group.py:1]: Just moved from tests dir

**Review by colinwkingen — APPROVED** [review]
LGTM. We discussed not enforcing the asset reference requirement in a subsequent PR.

**Pull request merged**